RecoLocalCalo/HGCalRecAlgos/interface/HGCalUncalibRecHitRecWeightsAlgo.h

```diff
@@ -33,23 +33,26 @@ class HGCalUncalibRecHitRecWeightsAlgo {
 
   void set_tdcOnsetfC(const double tdcOnset) { tdcOnsetfC_ = tdcOnset; }
 
+  void set_tofDelay(const double tofDelay) { tofDelay_ = tofDelay; }
+
   void set_fCPerMIP(const std::vector<double>& fCPerMIP) {
     if (std::any_of(fCPerMIP.cbegin(), fCPerMIP.cend(), [](double conv) { return conv <= 0.0; })) {
       throw cms::Exception("BadConversionFactor") << "At least one of fCPerMIP is zero!" << std::endl;
     }
     fCPerMIP_ = fCPerMIP;
   }
 
-  void setGeometry(const HGCalGeometry* geom) {
-    if (geom)
-      ddd_ = &(geom->topology().dddConstants());
+  void setGeometry(const edm::ESHandle<HGCalGeometry>& geom) {
```

Comment:
Minor: you could use `bool setGeometry` (instead of `void`), returning `geom_ != nullptr` (or similar), to give the user a handle to know if `geom_` was set or not.

Reply:
Thanks for the feedback. I implemented it in the new commit.